### PR TITLE
Added get /sandbox/:id endpoint

### PR DESCRIPTION
**Pull request by mishushakov**
- Changed spec.yml to add /sandbox/:id endpoint
- Added API method to retrieve sandbox by id

<details open>
<summary>:sparkles: <i><h3>Description by Callstackai</h3></i></summary>

This PR adds a new endpoint to retrieve a sandbox by its ID, including the necessary API method and updates to the OpenAPI specification.

<details>
<summary><b>Diagrams of code changes</b></summary>

```mermaid
sequenceDiagram
    participant Client
    participant API
    participant Handler
    participant Orchestrator
    participant Cache
    participant DB

    Client->>API: GET /sandboxes/{sandboxID}
    API->>Handler: GetSandboxesSandboxID()
    Handler->>Handler: Get Team Info from Context
    Handler->>Orchestrator: GetInstance(sandboxID)
    Orchestrator->>Cache: GetInstance(sandboxID)
    Cache-->>Handler: Instance Info
    
    alt Instance Found
        Handler->>DB: Query EnvBuild
        DB-->>Handler: Build Info
        Handler->>Handler: Construct RunningSandbox Response
        Handler-->>Client: 200 OK with Sandbox Details
    else Instance Not Found
        Handler-->>Client: 404 Not Found
    end
```

</details>

<details>
<summary><b>Files Changed</b></summary>
<table>
<tr><th>File</th><th>Summary</th></tr>
<tr><td><a href=https://github.com/e2b-dev/infra/pull/225/files#diff-bfa6ea3da4f29a1db196d3d485417366fd4bc3e73662eb6c011ae4a236246925>packages/api/internal/api/api.gen.go</a></td><td>Added the GetSandboxesSandboxID method to the ServerInterface and its wrapper.</td></tr>
<tr><td><a href=https://github.com/e2b-dev/infra/pull/225/files#diff-ae30b8e75db688f70c3d6d32afb639f3a10ef77f0ba75be27b5e440eb338615c>packages/api/internal/api/spec.gen.go</a></td><td>Updated the Swagger specification to include the new /sandboxes/{sandboxID} endpoint.</td></tr>
<tr><td><a href=https://github.com/e2b-dev/infra/pull/225/files#diff-1e9907fdfb817cf7f21f37de3d67c3dd7615cf164620128c0f8e3e8fee508cea>packages/api/internal/handlers/sandbox_get.go</a></td><td>Implemented the GetSandboxesSandboxID handler to retrieve sandbox details.</td></tr>
<tr><td><a href=https://github.com/e2b-dev/infra/pull/225/files#diff-5e643309b3191b00aa4ad91a75f3f357cc29cbfd02c897b6344805af7558b927>packages/api/internal/orchestrator/list_instances.go</a></td><td>Added the GetInstance method to the Orchestrator to fetch instance details.</td></tr>
<tr><td><a href=https://github.com/e2b-dev/infra/pull/225/files#diff-5b7a50ab5c4be5545227c7c591b86773a13143499d6d2f101df67776abdb847d>spec/openapi.yml</a></td><td>Defined the OpenAPI specification for the new endpoint.</td></tr>

</table>
</details>

*This PR includes files in programming languages that we currently do not support. We have not reviewed files with the extensions `.yml`. <a href=https://docs.callstack.ai/introduction>See list of supported languages.</a>*

</details>

### CODE REVIEW

File: packages/api/internal/handlers/sandboxes_list.go [hunk 1]
```diff
@@ -88,8 +88,65 @@ func (a *APIStore) GetSandboxes(c *gin.Context) {
...
 	// Sort sandboxes by start time descending
 	slices.SortFunc(sandboxes, func(a, b api.RunningSandbox) int {
-                return a.StartedAt.Compare(b.StartedAt)
+		return a.StartedAt.Compare(b.StartedAt)
 	})
 
 	c.JSON(http.StatusOK, sandboxes)
 }
+
+func (a *APIStore) GetSandboxesSandboxID(c *gin.Context, id string) {
```
Comment (jakubno) [inline packages/api/internal/handlers/sandboxes_list.go]: can you move to corresponding file?

**Review by jakubno — APPROVED** [review]

**Comment by jakubno** [comment]
Could you please  `dev` as a base branch? 🙏🏻

File: packages/api/internal/handlers/sandbox_get.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	telemetry.ReportEvent(ctx, "get running instance")
+
+	info, err := a.instanceCache.GetInstance(id)
+	if err != nil {
+		c.JSON(http.StatusNotFound, nil)
+		return
+	}
+
+	if *info.TeamID != team.ID {
+		c.JSON(http.StatusNotFound, nil)
```
Comment (jakubno) [inline packages/api/internal/handlers/sandbox_get.go]: This should probably be 403

File: packages/api/internal/handlers/sandbox_get.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	ctx := c.Request.Context()
+
+	teamInfo := c.Value(auth.TeamContextKey).(authcache.AuthTeamInfo)
+	team := teamInfo.Team
+
+	telemetry.ReportEvent(ctx, "get running instance")
+
+	info, err := a.instanceCache.GetInstance(id)
+	if err != nil {
+		c.JSON(http.StatusNotFound, nil)
```
Comment (jakubno) [inline packages/api/internal/handlers/sandbox_get.go]: Coudl you add some text explaining the error? 🙏🏻

File: packages/api/internal/handlers/sandbox_get.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	a.posthog.IdentifyAnalyticsTeam(team.ID.String(), team.Name)
+	properties := a.posthog.GetPackageToPosthogProperties(&c.Request.Header)
+	a.posthog.CreateAnalyticsTeamEvent(team.ID.String(), "get running instance", properties)
+
+	build, err := a.db.Client.EnvBuild.Query().Where(envbuild.ID(*info.BuildID)).First(ctx)
+	if err != nil {
+		telemetry.ReportCriticalError(ctx, err)
+
```
Comment (jakubno) [inline packages/api/internal/handlers/sandbox_get.go:43]: there's no status code set/response

**Review by jakubno — CHANGES_REQUESTED** [review]

File: packages/api/internal/handlers/sandbox_get.go (hunk 2 above)
Comment (mishushakov) [reply packages/api/internal/handlers/sandbox_get.go]: I think for security reasons we don't want the users to know the sandbox exists. For example sandboxes_list currently returns empty list if team has no access

File: packages/api/internal/handlers/sandbox_get.go (hunk 3 above)
Comment (mishushakov) [reply packages/api/internal/handlers/sandbox_get.go]: Yes!

File: packages/api/internal/handlers/sandbox_get.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	}
+
+	a.posthog.IdentifyAnalyticsTeam(team.ID.String(), team.Name)
+	properties := a.posthog.GetPackageToPosthogProperties(&c.Request.Header)
+	a.posthog.CreateAnalyticsTeamEvent(team.ID.String(), "get running instance", properties)
+
+	build, err := a.db.Client.EnvBuild.Query().Where(envbuild.ID(*info.BuildID)).First(ctx)
+	if err != nil {
+		telemetry.ReportCriticalError(ctx, err)
+		c.JSON(http.StatusInternalServerError, nil)
```
Comment (jakubno) [inline packages/api/internal/handlers/sandbox_get.go]: Could also add a message here
```suggestion
		c.JSON(http.StatusInternalServerError, nil)
```

**Review by jakubno — APPROVED** [review]

**Pull request merged**